README.md

```diff
@@ -294,6 +294,20 @@ type T = [/** foo */ number, /** bar */ string];
 The downlevel semantics are exactly the same as the original, but
 the TypeScript language service won't be able to show the member names.
 
+### `{ [key: T]: A }` (4.4)
+
+Typescript 4.4 supports index signature types other than `string`, `number`:
+
+```ts
+type O = { [key: T]: A };
+```
+
+becomes:
+
+```ts
+type O = { [key: string]: A };
+```
+
```

Comment:
Please add a Semantics section. Specifically, the transformation as written makes sense for cases where

1. the type is a template literal
2. the type is *reference* to a template literal
because `x${number}y -> string` (for example) is more permissive.

I'm not sure it makes sense for the following types that also work in 4.4:

3. `symbol`
4. a reference to `symbol`
5. unions of string, symbol, number or template literals.
6. a reference to unions of string, symbol, number or template literals.

cases (2), (4) and (6) are syntactically ambiguous, so I don't think they can be downlevelled with this tool.
(3) can't be downlevelled at all.
(5) can be downlevelled to multiple index signatures if the `symbol` isn't in the union.

Reply:
I've updated it to satisfy only the first case. 

Regarding making it compatible with the second case, can we add a flag for "unsafe" transformations or allow transformations provided by user?